Bottom: d64c0c6cd73d1cad23e3b86c4b61dfa23695c1c3
Top:    cb75e2edc7160a9760ad959fec5c970d70612c31
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-11 23:40:16 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 31021c9..b01a5ae 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1101,7 +1101,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
